### PR TITLE
docs - `hashi_vault`  migration guide

### PR DESCRIPTION
**Pull request by briantist**
##### SUMMARY

This will require (it will contain RST references to those plugins):
- #246

With the KV read content, we have full coverage of the functionality that `community.hashi_vault.hashi_vault` provided, and I can write a complete guide for making use of the newer content.

##### ISSUE TYPE

- Docs Pull Request

### CODE REVIEW

**Comment by briantist** [comment]
I'm also going to add a link to this guide in the `seealso:` section of the `hashi_vault` lookup but I'm going to wait until it's ready to merge, because that change will trigger the full CI to run, and I want to avoid that when fixing up typos and stuff in the guide.

**Review by felixfontein — APPROVED** [review]

**Comment by briantist** [comment]
thanks again for another review @felixfontein !

**Pull request merged**
